packages/material-ui/src/Tooltip/Tooltip.test.js

```diff
@@ -255,12 +255,23 @@ describe('<Tooltip />', () => {
     });
   });
 
-  it('should forward properties to the child element', () => {
-    const wrapper = shallow(
-      <Tooltip className="foo" {...defaultProps}>
-        <h1>H1</h1>
-      </Tooltip>,
-    );
-    assert.strictEqual(wrapper.find('h1').props().className, 'foo');
+  describe('forward', () => {
+    it('should forward properties to the child element', () => {
+      const wrapper = shallow(
+        <Tooltip className="foo" {...defaultProps}>
+          <h1>H1</h1>
+        </Tooltip>,
+      );
+      assert.strictEqual(wrapper.find('h1').props().className, 'foo');
+    });
+
+    it('should respect the properties priority', () => {
+      const wrapper = shallow(
+        <Tooltip className="foo" {...defaultProps}>
+          <h1 className="bar">H1</h1>
+        </Tooltip>,
+      );
+      assert.strictEqual(wrapper.find('h1').props().className, 'bar');
```

Comment:
Has nothing to do with this PR but this is why forwarding/merging props with `cloneElement` is always dangerous. At least by looking at it it might be obvious that `foo` will be ignored but you could also argue that this warrants a special merge strategy for the props (i.e. concatenating classes). If the child props are opaque then this will be less obvious e.g. the child is another component without passed props that then renders the h1 with a different class and you're then debugging why `foo` is not applied in the DOM.

Reply:
I agree, let's merge the class names. I believe it's what we do by "default" in the other components.